File: fsharp/space-age/SpaceAge.fs
module SpaceAge

type Planet = EarthYear of float

let Earth = EarthYear 1.
let Mercury = EarthYear 0.2408467
let Venus = EarthYear 0.61519726
let Mars = EarthYear 1.8808158
let Jupiter = EarthYear 11.862615
let Saturn = EarthYear 29.447498
let Uranus = EarthYear 84.016846
let Neptune = EarthYear 164.79132

let secondsInEarthYear = 31557600.

let age (EarthYear earthYear) (seconds: int64): float = 
    let round2 (x:float) = System.Math.Round (x, 2)
    float seconds / (earthYear * secondsInEarthYear) |> round2

File: go/space-age/space_age.go
package space

const earthSeconds = 31557600.0

// Planet is a named string type.
type Planet string

// Age returns the equivalent age on a given planet in that planet's years.
func Age(seconds float64, planet Planet) float64 {
	switch planet {
	case "Mercury":
		return seconds / earthSeconds / 0.2408467
	case "Venus":
		return seconds / earthSeconds / 0.61519726
	case "Mars":
		return seconds / earthSeconds / 1.8808158
	case "Jupiter":
		return seconds / earthSeconds / 11.862615
	case "Saturn":
		return seconds / earthSeconds / 29.447498
	case "Uranus":
		return seconds / earthSeconds / 84.016846
	case "Neptune":
		return seconds / earthSeconds / 164.79132
	default:
		return seconds / earthSeconds
	}
}
